[master] Fix MySQL cache entry enumeration output

jealouscloud opened the pull request:
### What does this PR do?

### What issues does this PR fix or reference?
Fixes: #66102 
This fixes minion detection via manage.present.

### Previous Behavior
```
$ salt-run  manage.present
$
```

### New Behavior
```
$ salt-run  manage.present
- minion1
- minion2
$
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [*] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

jealouscloud committed:
Fix entry enumeration output

jealouscloud committed:
Add changelog

jealouscloud committed:
pre-commit suggested fixes

jealouscloud committed:
Address result correctly

twangboy commented:
Looks like we have some test failures (pre-commit). Also, this needs to have a test written.

twangboy reviewed (changes requested):
Please write a test for this

dwoz committed:
Fix mysql_cache.ls() to return both direct keys and sub-bank names

The original PR #66103 fix (SELECT bank LIKE 'bank/%') correctly enumerated
sub-banks (fixing manage.present) but dropped direct keys (SELECT etcd_key
WHERE bank=exact), breaking the cross-backend cache contract exercised by
tests/pytests/functional/cache/helpers.py::run_common_cache_tests.

Replace with two queries whose results are union'd:
1. SELECT etcd_key WHERE bank=<bank>  — direct keys at this bank level
2. SELECT DISTINCT SUBSTRING_INDEX(SUBSTR(bank, N), '/', 1) WHERE bank LIKE
   '<bank>/%'  — first-level sub-bank names (MySQL SUBSTR is 1-indexed)

Add three unit tests covering direct keys, sub-bank enumeration, and
combined results.

Fixes #66102

twangboy commented:
The PR description doesn't seem to match what this PR is doing
